### PR TITLE
[infra-proxy-service] Add Environments gRPC & HTTP API

### PR DESCRIPTION
**Pull request by vsingh-msys**
### :nut_and_bolt: Description: What code changed, and why?

- Add infra environment list API with attributes name.

gRPC API:
```
chef-automate dev grpcurl infra-proxy-service chef.automate.domain.infra_proxy.service.InfraProxy.GetEnvironments -- -d '{"org_id": "8f1646d3-35ae-4562-84d7-ff013f3973f2", "server_id": "c7d36e02-4816-4bfa-a1b9-d239f56c9b8d"}'

```
HTTP API:
```
curl -sSkH "api-token: $TOK" https://a2-dev.test/api/v0/infra/servers/{server_id}/orgs/{org_id}/environments

Response format:

{
  "environments": [
    {
      "name": "_default"
    },
  ]
}
```

- Add infra environment detail API with attributes name, chef_type, description, run_list, json_class

gRPC API:
```
chef-automate dev grpcurl infra-proxy-service chef.automate.domain.infra_proxy.service.InfraProxy.GetEnvironment -- -d '{"org_id": "8f1646d3-35ae-4562-84d7-ff013f3973f2", "server_id": "c7d36e02-4816-4bfa-a1b9-d239f56c9b8d", "name": "_default"}'
```
HTTP API:
```
curl -sSkH "api-token: $TOK" https://a2-dev.test/api/v0/infra/servers/{server_id}/orgs/{org_id}/environments/name

Response format:

{
  "name": "_default",
  "chefType": "environment",
  "description": "The default Chef environment",
  "jsonClass": "Chef::Environment"
}
```

### :chains: Related Resources
#1544 
### :+1: Definition of Done
 - Able to access environments list and detail API
### :athletic_shoe: How to Build and Test the Change
 - `rebuild components/infra-proxy-service`
 - `rebuild components/automate-gateway`

In order to test these features, we can try to curl command to test the API describes in the description.

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
Signed-off-by: Vivek Singh <vivek.singh@msystechnologies.com>

### CODE REVIEW

File: components/infra-proxy-service/server/v1/environments.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid org id: %s", err.Error())
+	}
+
+	en, err := client.Environments.Get(req.Name)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	defaultAttributes, _ := json.Marshal(en.DefaultAttributes)
```
Comment (vsingh-msys) [inline components/infra-proxy-service/server/v1/environments.go]: Specific data type to hold `DefaultAttributes` data?

Comment (vsingh-msys) [reply components/infra-proxy-service/server/v1/environments.go]: Took the reference from `NodeAttribute` for dynamic attributes as:

https://github.com/chef/automate/blob/440f656fc4d396b40e1e70386e56d42e93f92e16/api/external/cfgmgmt/response/nodes.proto#L65-L72

File: api/interservice/infra_proxy/request/environments.proto [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    // Id of the Server.
+    string server_id = 2;
+}
+
+message Environment {
+    // Id of the Org.
+    string org_id    = 1;
+    // Id of the Server.
+    string server_id = 2;
+    // Name of the environment.
```
Comment (vjeffrey) [inline api/interservice/infra_proxy/request/environments.proto]: thanks for including the descriptions of the fields!!

File: components/infra-proxy-service/server/v1/environments.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	overrideAttributes, _ := json.Marshal(en.OverrideAttributes)
```
Comment (vjeffrey) [inline components/infra-proxy-service/server/v1/environments.go]: err check? same comment for a couple lines up too

Comment (vsingh-msys) [reply components/infra-proxy-service/server/v1/environments.go]: sure @vjeffrey thanks for pointing it out!

**Pull request merged**